src/devices/bus/mc10/mc10cart.cpp

```diff
@@ -0,0 +1,239 @@
+        17  A5                 34  GND
+
+	SEL is an input to the MC-10 that allows the cartridge to remove
+	the internal chips from the bus.
+
+*********************************************************************/
+
+#include "emu.h"
+#include "mc10cart.h"
+
+#include "mc10_mcx128.h"
+#include "mc10_pak.h"
+#include "mc10_ram.h"
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(MC10CART_SLOT, mc10cart_slot_device, "mc10cart_slot", "MC-10 Cartridge Slot")
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+//-------------------------------------------------
+//  mc10cart_slot_device - constructor
+//-------------------------------------------------
+mc10cart_slot_device::mc10cart_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock) :
+	device_t(mconfig, MC10CART_SLOT, tag, owner, clock),
+	device_single_card_slot_interface<device_mc10cart_interface>(mconfig, *this),
+	device_image_interface(mconfig, *this),
+	m_nmi_callback(*this),
+	m_cart(nullptr),
+	m_memspace(*this, finder_base::DUMMY_TAG, -1)
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void mc10cart_slot_device::device_start()
+{
+	m_nmi_callback.resolve();
+	m_cart = get_card_device();
+}
+
+//-------------------------------------------------
+//  install_bank - install RAM in host computer
+//-------------------------------------------------
+
+void mc10cart_slot_device::install_bank(offs_t start, offs_t end, uint8_t *data)
+{
+	m_memspace->install_ram(start, end, data);
+}
+
+//-------------------------------------------------
+//  install_bank - install ROM in host computer
+//-------------------------------------------------
+
+void mc10cart_slot_device::install_rom(offs_t start, offs_t end, uint8_t *data)
+{
+	m_memspace->install_rom(start, end, data);
+}
+
+//-------------------------------------------------
+//  set_nmi_line
+//-------------------------------------------------
+
+void mc10cart_slot_device::set_nmi_line(int state)
+{
+		if (!(m_nmi_callback).isnull())
+			(m_nmi_callback)(state);
+}
+
+//-------------------------------------------------
+//  call_load
+//-------------------------------------------------
+
+image_init_result mc10cart_slot_device::call_load()
+{
+	if (m_cart)
+	{
+		memory_region *cart_mem = m_cart->get_cart_memregion();
+		u8 *base = cart_mem->base();
+		offs_t read_length, cart_length = cart_mem->bytes();
+
+		if (!loaded_through_softlist())
+		{
+			read_length = fread(base, cart_length);
+		}
+		else
+		{
+			read_length = get_software_region_length("rom");
+			memcpy(base, get_software_region("rom"), read_length);
```

Comment:
Why is the copy necessary?  If it’s loaded from the software list, you’ll already have the content in a memory region.  You shouldn’t be doing unnecessary copies.

Reply:
Got rid of the copy.